### PR TITLE
Update dependencies for VS

### PR DESCRIPTION
**Pull request by cartermp**
This does a rather large update to our VS dependencies so that we can take recent Roslyn updates (recent meaning like, in the past year). It also sets us up to keep updating and get access to things like Inline Hints: https://github.com/dotnet/roslyn/pull/48529

Now we get to see if CI agrees with me

### CODE REVIEW

**Comment by cartermp** [comment]
noooo I got this to work on my machine :(

**Comment by cartermp** [comment]
`System.Exception : unable to find assembly Microsoft.VisualStudio.Editor.Implementation.dll`

Ugh, this is gonna suck to untangle. The VS Salsa mocks assume an implementation .dll that isn't shipped publicly

**Comment by cartermp** [comment]
lmao it worked 

Salsa just had a dumb dependency for no reason at all

File: vsintegration/tests/UnitTests/VisualFSharp.UnitTests.fsproj [hunk 1]
```diff
@@ -205,7 +205,8 @@
     <PackageReference Include="System.ValueTuple" Version="$(SystemValueTupleVersion)" />
     <PackageReference Include="VSSDK.VSHelp" Version="$(VSSDKVSHelpVersion)" />
     <PackageReference Include="VSSDK.VSLangProj.8" Version="$(VSSDKVSLangProj8Version)" PrivateAssets="all" ExcludeAssets="contentFiles;build;analyzers;native" />
-	<PackageReference Include="StreamJsonRpc" Version="$(StreamJsonRpcVersion)" />
+	  <PackageReference Include="StreamJsonRpc" Version="$(StreamJsonRpcVersion)" />
```
Comment (cartermp) [inline vsintegration/tests/UnitTests/VisualFSharp.UnitTests.fsproj:208]: I need to build a hive of VS to check if the settings this deals with work

File: tests/fsharp/app.config [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
       <dependentAssembly>
         <assemblyIdentity name="System.Collections.Immutable" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
-        <bindingRedirect oldVersion="0.0.0.0-1.2.3.0" newVersion="1.2.3.0" />
+        <bindingRedirect oldVersion="0.0.0.0-5.0.0.0" newVersion="5.0.0.0" />
```
Comment (cartermp) [inline tests/fsharp/app.config:7]: I'm gonna see if these are even needed in the first place

**Comment by cartermp** [comment]
Now for the really spicy meatball: can I just delete some bindingredirect nonsense?

Let's ask CI!

File: eng/Versions.props [hunk 3]
```diff
@@ -107,69 +109,83 @@
...
+    <MicrosoftVisualStudioUtilitiesVersion>16.7.30329.38</MicrosoftVisualStudioUtilitiesVersion>
+    <MicrosoftVisualStudioValidationVersion>16.8.2-alpha</MicrosoftVisualStudioValidationVersion>
     <MicrosoftVisualStudioWCFReferenceInteropVersion>9.0.30729</MicrosoftVisualStudioWCFReferenceInteropVersion>
     <MicrosoftVSSDKBuildToolsVersion>16.5.2044</MicrosoftVSSDKBuildToolsVersion>
     <VSSDKDebuggerVisualizersVersion>12.0.4</VSSDKDebuggerVisualizersVersion>
     <VSSDKVSLangProjVersion>7.0.4</VSSDKVSLangProjVersion>
     <VSSDKVSLangProj8Version>8.0.4</VSSDKVSLangProj8Version>
     <VSSDKVSLangProj11Version>11.0.4</VSSDKVSLangProj11Version>
     <VSSDKVSHelpVersion>7.0.4</VSSDKVSHelpVersion>
+
```
Comment (brettfo) [inline eng/Versions.props:183]: FYI, Arcade removes empty lines from this file, so it'll only be pretty for a few days.

Comment (cartermp) [reply eng/Versions.props:183]: well shit

**Review by brettfo — APPROVED** [review]
Your erratic whitespace makes me want to `rm -rf @brettfo`, but otherwise it looks good.

**Comment by cartermp** [comment]
The whitespace is bonkers - editing in VSCode and VS apparently created inconsistent spacing. I hate so much the giant baby.

**Comment by cartermp** [comment]
I'll just merge this (before I do something stupid) and clean it up in another PR where I update again to allow us to use inline hints.

**Pull request merged**